subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/ResolveTaskArtifactStateTaskExecuter.java

```diff
@@ -48,12 +48,6 @@ public void execute(TaskInternal task, TaskStateInternal state, TaskExecutionCon
         context.setTaskArtifactState(taskArtifactState);
         outputs.setHistory(taskArtifactState.getExecutionHistory());
         LOGGER.info("Putting task artifact state for {} into context took {}.", task, clock.getElapsed());
-        try {
-            executer.execute(task, state, context);
-        } finally {
-            outputs.setHistory(null);
-            context.setTaskArtifactState(null);
-            LOGGER.debug("Removed task artifact state for {} from context.");
-        }
```

Comment:
this is now moot given we went back to the old decorator pattern, with a subsection of the task executers being wrapped in a build operation